feat: Add student guide (#283)
open-sauced/docs: docs/opensauced-guides/students-guide/students-guide.md

4. Manage your email preferences.

### Your Profile

Your profile is a great resource for showcasing your contributions and sharing your open source work with others.

![your profile](../../../static/img/profile.png)

On your profile, you will see the "Highlights" tab, which you can learn more about [in this section](#highlights-connecting-and-recognizing-contributions). Other tabs you may find useful on your profile are: 

- **Contributions**: This tab includes a graph showing the number of PRs you have opened, [the velocity of your PRs](../../welcome/glossary.md), and the number of repositories you have contributed to in the last thirty days. Underneath the graph is a "Latest PRs" chart, which shows a list of current and recent PRs. This tab is helpful if you want to showcase a steady and consistent open source contribution experience.
- **Recommendations**: This tab includes a list of open source projects that are chosen based on the programming languages you picked in the "Interests" section of your public profile. It is useful if you're looking for new projects to contribute to.

:::tip

> Don't just share your profile, include in on things like your resume, LinkedIn, your GitHub Readme, and  your personal website to showcase your most up-to-date open source contributions.

Author: No. I mean don't just share it on social media. Add the link to your OpenSauced Profile on your resume, LI, GH Readme, and site. I'll work on the wording.

:::

### Your Dev Card

> Another fun way to share your open source contributions is through your [Dev Card](../../features/dev-card.md). Your Dev Card is a visual representation of your open source contributions. It includes your profile picture, name, bio, and a list of your latest contributions. 

Reviewer: I've read through one more time, a nd this one is the last thing I've found. 

```suggestion
Another fun way to share your open source contributions is through your [Dev Card](../../features/dev-card.md). Your Dev Card is a visual representation of your open source contributions. It includes your profile picture, name, bio, and the number of your latest contributions. 
```

#### Highlighting PRs

1. **Describe the PR**: Start with a clear summary of what your PR accomplishes. For example, "I added a new feature to the project's Chrome Extension that allows users to summarize content."
2. **Highlight your solution and its impact**: Briefly describe the solution, any challenges you faced, or any part of the PR you're particularly proud of. For example, "Initially, I faced challenges with the API, but I was able to resolve them using a different endpoint."
3. **Reflect on your experience**: Mention what you learned from working on the PR. For example, "I learned a lot about the Chrome Extension API and how to use it to create new features."

#### Highlighting Dev.to Blog Posts

1. **Describe the topic**: Begin with an interesting blog post summary to help others to understand your writing.
2. **Discuss the inspiration**: Explain why you wrote this blog post. Is it to guide others on improving their coding skills, raise awareness about an issue in the tech community, or display your progress in learning a new programming language?
3. **End with a call-to-action**: Encourage others to read and engage with the post.

## Creating Your Workspace

> As a student working on multiple projects or contributing to various open source repositories, keeping track of progress, goals, and collaboration can be daunting. To keep track of the repositories that are most important to you or repositories that you're working on with a team, OpenSauced Workspaces can be your go-to tool for organizing and understanding your open source journey.

Author: I think the , is fine where it is bc it's modifying student here, describing the kind of student.

- **Enhance team collaboration**: If you're working with a group you can keep track of what everyone is working on. This transparency is key to teamwork and support, whether you're setting goals for a class project, contributing to a group assignment, or working together for accountability.
- **Deepen your project insight**: Understanding the underlying activities of the projects you're tracking helps you to understand their progress, health, and the community of student contributors you're part of.
- **Celebrate and communicate your achievements**: When it's time to talk about your experience, having analytics on the projects you worked on allows you to showcase your contributions, acknowledge the efforts of your peers, and clearly communicate the hard work invested.

Workspaces offer a centralized platform that’s perfect for student's to organize, track, and analyze the open source projects they're working on. 

### Creating a New Workspace

To create a workspace, start by accessing the **Workspace switcher** in the sidebar. From there, you can create a new workspace or view an existing one.

![workspace switcher](../../../static/img/workspace-switcher.png)

<br/>

:::note

> Whenever you select "Home", you'll be taken to the last workspace you opened.

Author: Normally that , would be inside the quotes, but for clarity here bc it's referring to a specific word they need to see, I think it's clearer to have it outside.

3. **Coursework Management**: For managing group open source projects or individual assignments, a workspace can be used to organize repositories that are part of the assignments, keeping all relevant repositories in one accessible location.

4. **Skill-Specific Exploration**: You can use workspaces to follow repositories that help you build specific skills or understand certain technologies, creating a tailored learning environment that aligns with your career path or focus.

5. **Community Engagement Tracking**: By adding repositories you're interested in, you can monitor and engage with the communities behind them, understanding the dynamics of open source collaboration, which is invaluable for both personal and professional networking.

## Repository Insights: Connecting Your Repositories

OpenSauced [Repository Insights](../../features/repo-insights.md) allows you to track open source projects you are interested in. You can use Repository Insights Pages to track repositories' growth, analyze work, and connect with others.

> ### Why Create a Repository Insights Page

Reviewer: ```suggestion
### Why Create a Repository Insights Page?
```

### Creating a New Repository Insight Page

To track the repositories, click the "+" next to "Insights" in the sidebar, then select "New Repository Insight." You will be redirected to a page where you can create your new Repository Insight Page.

There are two ways to add repositories to your Repository Insight Page:

1. **Sync GitHub organization**: Syncing your GitHub organizations is a good idea if you want to keep track of their open source project activities and trends.
2. **Connect individual repositories**: Connecting individual repositories is a good idea if you want to keep track of the contributions of a specific repository or group of repositories.

![sync Repository Insights gif](../../../static/gif/insight-page-demo.gif)

### Using Your Repository Insights Page

Once you have connected your repositories, you will be redirected to your Repository Insight Page. Here, you will see a dashboard with an overview of the repositories and the contributors who have contributed to them.

There are three tabs that provide you with more information on these repositories:

#### Repositories Dashboard

> If you're looking for an active project to contribute to, this tab can help you as it contains information such as activity levels, PR overview, PR velocity, spam PRs, contributors, and activity stats over the last 30 days. This information is also a good way to connect with others on the project or where to find support.

Reviewer: We can split the first sentence to understand it better.

```suggestion
If you're looking for an active project to contribute to, this tab can help you. It contains information such as activity levels, PR overview, PR velocity, spam PRs, contributors, and activity stats over the last 30 days. This information is also a good way to connect with others on the project or where to find support.
```

![contributors insight list](../../../static/img/contributors-insights.png)

:::tip

Want to learn more about your peers? You can select and add them to a Contributor Insight Page.

:::

#### Activity

The "Activity" tab shows a graph of the last time each contributor created their PR and how many lines of code they've touched. It also provides more detailed information on each contributor when you hover over their image, including their latest PRs and repositories they've contributed to. Clicking their image will bring you to their profile on OpenSauced.

![activity graph](../../../static/img/activity-repos-insights.png)

Beyond understanding repositories, you may want to get insights into other contributors to help you learn more about how others are contributing to open source. One way to do that is through Contributor Insights.

## Contributor Insights: Connecting with Contributors

The [Contributor Insights](../../features/contributor-insights.md) feature helps you connect with other contributors or learn more about what others are up to in open source. With this feature, you can categorize, watch, and check out different groups of contributors within open source projects.

> ### What You can do with Contributor Insights?

Reviewer: Not sure if we should have a question mark here.

```suggestion
### What You Can Do with Contributor Insights
```

- **Explore**: The "Explore" tab is a great place to find new open source projects to contribute to.
- **Highlights**: Leveraging OpenSauced's Highlights feature is a great way for you to showcase your valuable contributions to open source projects.
- **Repository Insights**: OpenSauced Repository Insights feature allows you to track open source projects you are interested in.
- **Contributor Insights**: The Contributor Insights feature helps you understand other contributors or track your own work in open source.

## Conclusion

Embarking on your open source journey with OpenSauced opens up a world of learning and development opportunities. It's a chance to apply your knowledge to real-world projects, collaborate with a global community, and build a portfolio that stands out to educators and employers alike. We're excited to support you on this journey and can't wait to see the contributions you'll make and the milestones you'll achieve. If you have any questions or need assistance, our community is here to help you navigate your path in open source.

Here are some other resources you might find helpful in your journey:

- [100 Days of OSS](../../community/100-days-of-oss.md)

> - [Intro to Open Source](https://intro.opensauced.pizza/#/)

Reviewer: ```suggestion
- [Intro to Open Source Course](https://intro.opensauced.pizza/#/)
```
